cns/restserver/util.go

```diff
@@ -299,9 +331,35 @@ func (service *HTTPRestService) attachOrDetachHelper(req cns.ConfigureContainerN
 			Message:    "[Azure CNS] Error. NetworkContainerid is empty"}
 	}
 
+	var (
+		dncEP        = service.GetOption(acn.OptPrivateEndpoint).(string)
+		infraVnet    = service.GetOption(acn.OptInfrastructureNetwork).(string)
+		nodeID       = service.GetOption(acn.OptNodeID).(string)
+		isManagedDnc = service.GetOption(acn.OptManaged).(bool)
+	)
+
 	existing, ok := service.getNetworkContainerDetails(cns.SwiftPrefix + req.NetworkContainerid)
+	if isManagedDnc && operation == attach {
+		if ok {
+			if existing.CreateNetworkContainerRequest.WaitingForUpdate {
+				_, returnCode, message := isNCWaitingForUpdate(existing.CreateNetworkContainerRequest.Version, req.NetworkContainerid)
+				if returnCode != Success {
+					return cns.Response{
+						ReturnCode: returnCode,
+						Message:    message}
+				}
+			}
+		} else {
+			returnCode, msg := service.SyncNodeStatus(dncEP, infraVnet, nodeID, json.RawMessage{})
+			if returnCode != Success {
+				return cns.Response{
+					ReturnCode: returnCode,
+					Message:    msg}
+			}
 
-	if !ok {
+			existing, _ = service.getNetworkContainerDetails(cns.SwiftPrefix + req.NetworkContainerid)
```

Comment:
dont we need to call isNCWaitingForUpdate after this?

Reply:
no it's already called in SyncNodeStatus